host/tests/unit/search-index-test.ts

```diff
@@ -241,8 +239,226 @@ module('Unit | search-index', function (hooks) {
     }
   });
 
-  // How would the searchDoc look for a cycle in the fields?
-  skip('can index a field with a cycle in the linksTo field');
+  test('can index a field with a cycle in the linksTo field', async function (assert) {
```

Comment:
need test for deserialization to assert === equality for the same instance that may appear in different places within instance values